net/client.go

```diff
@@ -107,13 +106,12 @@ func (s *server) pushLog(ctx context.Context, id thread.ID, lg thread.LogInfo, p
 
 	log.Debugf("pushing log %s to %s...", lg.ID, pid)
 
-	cctx, cancel := context.WithTimeout(ctx, reqTimeout)
-	defer cancel()
-	conn, err := s.dial(cctx, pid, grpc.WithInsecure())
+	client, err := s.dial(pid)
 	if err != nil {
 		return fmt.Errorf("dial %s failed: %s", pid, err)
 	}
-	client := pb.NewServiceClient(conn)
+	cctx, cancel := context.WithTimeout(ctx, reqTimeout)
```

Comment:
maybe you can export reqTimeout(and maybe split PushTime/PullTimeout) so we can play with it?